spec/npm/asciidoctor-all.spec.js

```diff
@@ -21,5 +37,76 @@ describe('Node.js', function () {
       expect(doc.attributes.$fetch('docname')).toBe('test.adoc');
     });
   });
+
+  describe('Converting file', function() {
+    it('should be able to convert a file', function() {
+      var expectFilePath = __dirname + '/test.adoc.html';
+      removeFile(expectFilePath);
+      try {
+        var doc = Asciidoctor.$convert_file(__dirname + '/test.adoc', null);
+        expect(fileExists(expectFilePath)).toBe(true);
+        var content = fs.readFileSync(expectFilePath, 'utf8');
+        expect(content).toContain('Hello world');
+      } finally {
+        removeFile(expectFilePath);
+      }
+    });
+
+    it('should be able to convert a file with custom css', function() {
+      var expectFilePath = __dirname + '/test.adoc.html';
+      removeFile(expectFilePath);
+      try {
+        var options = Opal.hash({'attributes': ['stylesheet=simple.css', 'stylesdir=css']});
+        var doc = Asciidoctor.$convert_file(__dirname + '/test.adoc', options);
+        expect(fileExists(expectFilePath)).toBe(true);
+        var content = fs.readFileSync(expectFilePath, 'utf8');
+        expect(content).toContain('css/simple.css');
+      } finally {
+        removeFile(expectFilePath);
+      }
+    });
+
+    it('should be able to convert a file with custom css embedded', function() {
+      var expectFilePath = __dirname + '/test.adoc.html';
+      removeFile(expectFilePath);
+      try {
+        var options = Opal.hash({'safe': 'server', 'attributes': ['stylesheet=simple.css', 'stylesdir=css']});
```

Comment:
To be more accurate, the default is to link to the CSS if safe mode is secure and embed the CSS if safe mode is server or below. When safe mode is server or below, you can still control this behavior using the linkcss attribute.